theme/icons_test.go

```diff
@@ -1,36 +1,489 @@
 	assert.NotEqual(t, content, cancel.Content())
 }
 
-func TestNewThemedResource(t *testing.T) {
+func TestNewThemedResource_TwoStaticResourceSupport(t *testing.T) {
+	fyne.CurrentApp().Settings().SetTheme(DarkTheme())
+	custom := NewThemedResource(helperNewStaticResource(), helperNewStaticResource())
+	content := custom.Content()
+	nm := custom.Name()
+
+	fyne.CurrentApp().Settings().SetTheme(LightTheme())
+	assert.NotEqual(t, content, custom.Content())
+	assert.Equal(t, nm, custom.Name())
+}
+
+func TestNewThemedResource_OneStaticResourceSupport(t *testing.T) {
 	fyne.CurrentApp().Settings().SetTheme(DarkTheme())
-	custom := NewThemedResource(mailcomposeDark, mailcomposeLight)
+	custom := NewThemedResource(helperNewStaticResource(), nil)
 	content := custom.Content()
+	nm := custom.Name()
 
 	fyne.CurrentApp().Settings().SetTheme(LightTheme())
 	assert.NotEqual(t, content, custom.Content())
+	assert.Equal(t, nm, custom.Name())
+}
+
+func TestThemedResource_Name(t *testing.T) {
+	sr := fyne.NewStaticResource("cancel_Paths.svg",
+		helperLoadBytes(t, "cancel_Paths.svg"))
+	dr := &ThemedResource{
+		source: sr,
+	}
+	assert.Equal(t, sr.Name(), dr.Name())
+}
+
+func TestThemedResource_Content_NoGroupsFile(t *testing.T) {
+	fyne.CurrentApp().Settings().SetTheme(DarkTheme())
+	sr := fyne.NewStaticResource("cancel_Paths.svg",
+		helperLoadBytes(t, "cancel_Paths.svg"))
+	dr := &ThemedResource{
+		source: sr,
+	}
+	assert.NotEqual(t, sr.Content(), dr.Content())
+}
+
+func TestThemedResource_Content_GroupPathFile(t *testing.T) {
+	fyne.CurrentApp().Settings().SetTheme(DarkTheme())
+	sr := fyne.NewStaticResource("check_GroupPaths.svg",
+		helperLoadBytes(t, "check_GroupPaths.svg"))
+	dr := &ThemedResource{
+		source: sr,
+	}
+	assert.NotEqual(t, sr.Content(), dr.Content())
+}
+
+func TestThemedResource_Content_GroupRectFile(t *testing.T) {
+	fyne.CurrentApp().Settings().SetTheme(DarkTheme())
+	sr := fyne.NewStaticResource("info_GroupRects.svg",
+		helperLoadBytes(t, "info_GroupRects.svg"))
+	dr := &ThemedResource{
+		source: sr,
+	}
+	assert.NotEqual(t, sr.Content(), dr.Content())
+}
+
+func TestThemedResource_Content_GroupPolygonsFile(t *testing.T) {
+	fyne.CurrentApp().Settings().SetTheme(DarkTheme())
+	sr := fyne.NewStaticResource("warning_GroupPolygons.svg",
+		helperLoadBytes(t, "warning_GroupPolygons.svg"))
+	dr := &ThemedResource{
+		source: sr,
+	}
+	assert.NotEqual(t, sr.Content(), dr.Content())
+}
+
+// a black svg object omits the fill tag, this checks it it still properly updated
+func TestThemedResource_Content_BlackFillIsUpdated(t *testing.T) {
+	fyne.CurrentApp().Settings().SetTheme(DarkTheme())
+	sr := fyne.NewStaticResource("cancel_PathsBlackFill.svg",
+		helperLoadBytes(t, "cancel_PathsBlackFill.svg"))
+	dr := &ThemedResource{
+		source: sr,
+	}
+	assert.NotEqual(t, sr.Content(), dr.Content())
+}
+
+// Test Asset Sources
+func Test_FyneLogo(t *testing.T) {
+	fyne.CurrentApp().Settings().SetTheme(DarkTheme())
```

Comment:
In every one of these the theme setting does not seem to add anything to the test - can they be removed?

Reply:
my thinking was that we'd be sure that both themes were using the same resource, but, we have a test that checks that explicitly, so I agree it can be cleaned up

Reply:
cool